manim/scene/scene.py

```diff
@@ -519,6 +519,50 @@ def remove(self, *mobjects: Mobject):
                 self.restructure_mobjects(mobjects, list_name, False)
             return self
 
+    def replace(self, old_mobject: Mobject, new_mobject: Mobject) -> None:
+        """Replace one Mobject in the scene with another, preserving draw order.
+
+        If old_mobject is a submobject of some other Mobject (e.g. a Group),
+        the new_mobject will replace it inside the group, without otherwise
+        changing the parent mobject.
+
+        Parameters
+        ----------
+        old_mobject - A Mobject which must be in the scene. This method asserts if
+                      old_mobject is not in the scene
+        new_mobject - A Mobject which must not already be in the scene.
+
+        """
+        assert old_mobject is not None
+        assert new_mobject is not None
+
+        def replace_in_list(
+            mobj_list: list[Mobject], old_m: Mobject, new_m: Mobject
+        ) -> bool:
+            # We use breadth-first search because some Mobjects get very deep and
+            # we expect top-level elements to be the most common targets for replace.
+            for i in range(0, len(mobj_list)):
+                # Is this the old mobject?
+                if mobj_list[i] == old_m:
+                    # If so, write the new object to the same spot and stop looking.
+                    mobj_list[i] = new_m
+                    return True
+            # Now check all the children of all these mobs.
+            for mob in mobj_list:
```

Comment:
Allowing flake8 to be unhappy.

```suggestion
            for mob in mobj_list:  # noqa: SIM110
```